src/MobileDetect.php

```diff
@@ -1433,6 +1428,8 @@ public function isMobile(): bool
                 $this->cache->set($cacheKey, $result, $this->config['cacheTtl']);
                 return $result;
             }
+        } catch (CacheException $e) {
+            throw new MobileDetectException("Cache problem in isMobile(): {$e->getMessage()}");
```

Comment:
this should probably attach `$e` as previous exception to make debugging easier.